Digital Credential API: DigitalCredential should inherit from Credential interface

marcoscaceres opened the pull request:
#### 4882b74baf008ae9c3541cc569d907d0f553634f
<pre>
Digital Credential API: DigitalCredential should inherit from Credential interface
<a href="https://bugs.webkit.org/show_bug.cgi?id=268145">https://bugs.webkit.org/show_bug.cgi?id=268145</a>
<a href="https://rdar.apple.com/121643028">rdar://121643028</a>

Reviewed by Chris Dumez.

Implements inheriting Credential interface for DigitalCredential.

* LayoutTests/http/wpt/credential-management/setDigitalCredentialsEnabled.https-expected.txt: Added.
* LayoutTests/http/wpt/credential-management/setDigitalCredentialsEnabled.https.html: Added.
* Source/WebCore/Modules/credentialmanagement/BasicCredential.cpp:
(WebCore::BasicCredential::type const):
* Source/WebCore/Modules/credentialmanagement/BasicCredential.h:
* Source/WebCore/Modules/credentialmanagement/DigitalCredential.cpp:
(WebCore::DigitalCredential::DigitalCredential):
* Source/WebCore/Modules/credentialmanagement/DigitalCredential.h:
(WebCore::DigitalCredential::data const): Deleted.
* Source/WebCore/Modules/credentialmanagement/DigitalCredential.idl:
* Source/WebCore/bindings/js/JSBasicCredentialCustom.cpp:
(WebCore::toJSNewlyCreated):

Canonical link: <a href="https://commits.webkit.org/273864@main">https://commits.webkit.org/273864@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/48438a268f3bb04ee6af4c5b5ae6a1f3e9b559d1

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/36761 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/15696 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/39001 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/39403 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/32932 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/38006 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/18183 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/12809 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/31510 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/37323 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/13230 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/32489 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/11571 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/11580 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/32767 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/40651 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/33308 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/33110 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/37503 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/11860 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/9676 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/35623 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/13523 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/32447 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/8364 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/12258 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/12756 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

cdumez commented on Source/WebCore/Modules/credentialmanagement/DigitalCredential.idl:
```diff
@@ -26,7 +26,8 @@
-] interface DigitalCredential {
+    EnabledBySetting=DigitalCredentialsEnabled,
+    Conditional=WEB_AUTHN,
+] interface DigitalCredential : BasicCredential {
```
This change is web-facing so we should have a test for it.

marcoscaceres replied:
I added just a simple existence test, as I figure we will start adding actual WPTests once the spec starts getting padded out.

cdumez commented on LayoutTests/http/wpt/credential-management/setDigitalCredentialsEnabled.https.html:
```diff
@@ -0,0 +1,39 @@
+            log("FAIL: navigator.credentials.requestIdentity() must be exposed. Was enabled by pref.");
+        }
+
+        if (window.DigitalCredential) {
```
The PR says "DigitalCredential should inherit from Credential interface" so I think the test should at least cover that?

marcoscaceres replied:
Yeah, good call. Hopefully the updated test is sufficient.

cdumez reviewed (approved):
LGTM

marcoscaceres committed:
Digital Credential API: DigitalCredential should inherit from Credential interface
https://bugs.webkit.org/show_bug.cgi?id=268145
rdar://121643028

Reviewed by Chris Dumez.

Implements inheriting Credential interface for DigitalCredential.

* LayoutTests/http/wpt/credential-management/setDigitalCredentialsEnabled.https-expected.txt: Added.
* LayoutTests/http/wpt/credential-management/setDigitalCredentialsEnabled.https.html: Added.
* Source/WebCore/Modules/credentialmanagement/BasicCredential.cpp:
(WebCore::BasicCredential::type const):
* Source/WebCore/Modules/credentialmanagement/BasicCredential.h:
* Source/WebCore/Modules/credentialmanagement/DigitalCredential.cpp:
(WebCore::DigitalCredential::DigitalCredential):
* Source/WebCore/Modules/credentialmanagement/DigitalCredential.h:
(WebCore::DigitalCredential::data const): Deleted.
* Source/WebCore/Modules/credentialmanagement/DigitalCredential.idl:
* Source/WebCore/bindings/js/JSBasicCredentialCustom.cpp:
(WebCore::toJSNewlyCreated):

Canonical link: https://commits.webkit.org/273864@main
